$MirOS: ports/www/php/patches/patch-ext_standard_datetime_c,v 1.1.7.1 2005/03/18 16:00:00 tg Exp $
--- ext/standard/datetime.c.orig	Tue Aug 30 09:15:36 2005
+++ ext/standard/datetime.c	Wed Jan 25 17:56:14 2006
@@ -446,7 +446,7 @@ static void php_date(INTERNAL_FUNCTION_P
 				strcat(Z_STRVAL_P(return_value), day_full_names[ta->tm_wday]);
 				break;
 			case 'Y':		/* year, numeric, 4 digits */
-				sprintf(tmp_buff, "%d", ta->tm_year + YEAR_BASE);  /* SAFE */
+				sprintf(tmp_buff, "%d", (int)ta->tm_year + YEAR_BASE);  /* SAFE */
 				strcat(Z_STRVAL_P(return_value), tmp_buff);
 				break;
 			case 'M':		/* month, textual, 3 letters */
@@ -460,7 +460,7 @@ static void php_date(INTERNAL_FUNCTION_P
 				strcat(Z_STRVAL_P(return_value), tmp_buff);
 				break;
 			case 'y':		/* year, numeric, 2 digits */
-				sprintf(tmp_buff, "%02d", ((ta->tm_year)%100));  /* SAFE */
+				sprintf(tmp_buff, "%02d", (int)((ta->tm_year)%100));  /* SAFE */
 				strcat(Z_STRVAL_P(return_value), tmp_buff);
 				break;
 			case 'm':		/* month, numeric */
@@ -586,7 +586,7 @@ static void php_date(INTERNAL_FUNCTION_P
 					day_short_names[ta->tm_wday],
 					ta->tm_mday,
 					mon_short_names[ta->tm_mon],
-					ta->tm_year + YEAR_BASE,
+					(int)ta->tm_year + YEAR_BASE,
 					ta->tm_hour,
 					ta->tm_min,
 					ta->tm_sec,
@@ -599,7 +599,7 @@ static void php_date(INTERNAL_FUNCTION_P
 					day_short_names[ta->tm_wday],
 					ta->tm_mday,
 					mon_short_names[ta->tm_mon],
-					ta->tm_year + YEAR_BASE,
+					(int)ta->tm_year + YEAR_BASE,
 					ta->tm_hour,
 					ta->tm_min,
 					ta->tm_sec,
@@ -613,7 +613,7 @@ static void php_date(INTERNAL_FUNCTION_P
 			case 'c':
 #if HAVE_TM_GMTOFF
 				sprintf(tmp_buff, "%04d-%02d-%02dT%02d:%02d:%02d%c%02d:%02d",
-					ta->tm_year + YEAR_BASE,
+					(int)ta->tm_year + YEAR_BASE,
 					ta->tm_mon + 1,
 					ta->tm_mday,
 					ta->tm_hour,
@@ -625,7 +625,7 @@ static void php_date(INTERNAL_FUNCTION_P
 				);
 #else
 				sprintf(tmp_buff, "%04d-%02d-%02dT%02d:%02d:%02d%c%02d:%02d",
-					ta->tm_year + YEAR_BASE,
+					(int)ta->tm_year + YEAR_BASE,
 					ta->tm_mon + 1,
 					ta->tm_mday,
 					ta->tm_hour,
@@ -949,14 +949,14 @@ char *php_std_date(time_t t TSRMLS_DC)
 				day_short_names[tm1->tm_wday],
 				tm1->tm_mday,
 				mon_short_names[tm1->tm_mon],
-				tm1->tm_year + 1900,
+				(int)tm1->tm_year + 1900,
 				tm1->tm_hour, tm1->tm_min, tm1->tm_sec);
 	} else {
 		snprintf(str, 80, "%s, %02d-%s-%02d %02d:%02d:%02d GMT",
 				day_full_names[tm1->tm_wday],
 				tm1->tm_mday,
 				mon_short_names[tm1->tm_mon],
-				((tm1->tm_year) % 100),
+				(int)((tm1->tm_year) % 100),
 				tm1->tm_hour, tm1->tm_min, tm1->tm_sec);
 	}
 	
